backend/src/main/kotlin/fr/gouv/cacem/monitorenv/domain/use_cases/missions/CreateOrUpdateMissionWithAttachedReporting.kt

```diff
@@ -0,0 +1,108 @@
+@file:Suppress("ktlint:standard:package-name")
+
+package fr.gouv.cacem.monitorenv.domain.use_cases.missions
+
+import fr.gouv.cacem.monitorenv.config.UseCase
+import fr.gouv.cacem.monitorenv.domain.entities.mission.*
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.ActionTypeEnum
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.EnvActionNoteEntity
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.EnvActionSurveillanceEntity
+import fr.gouv.cacem.monitorenv.domain.entities.mission.envAction.envActionControl.EnvActionControlEntity
+import fr.gouv.cacem.monitorenv.domain.repositories.IDepartmentAreaRepository
+import fr.gouv.cacem.monitorenv.domain.repositories.IFacadeAreasRepository
+import fr.gouv.cacem.monitorenv.domain.repositories.IMissionRepository
+import fr.gouv.cacem.monitorenv.domain.repositories.IReportingRepository
+import fr.gouv.cacem.monitorenv.domain.use_cases.missions.dtos.MissionDTO
+import fr.gouv.cacem.monitorenv.infrastructure.api.adapters.bff.inputs.missions.EnvActionAttachedToReportingIds
+
+@UseCase
+class CreateOrUpdateMissionWithAttachedReporting(
+    private val departmentRepository: IDepartmentAreaRepository,
+    private val facadeRepository: IFacadeAreasRepository,
+    private val missionRepository: IMissionRepository,
+    private val reportingRepository: IReportingRepository,
```

Comment:
```suggestion
    private val reportingRepository: IReportingRepository,
    private val createOrUpdateMission: CreateOrUpdateMission,
```


Je pense que `CreateOrUpdateMissionWithAttachedReporting` devrait utiliser le use-case `CreateOrUpdateMission` en dépendance injectée, pour éviter de répéter du code et pour faciliter le refacto par la suite.
Ce use-case deviendra plus simple, avec seulement les l.96 à l.106.